src/site_upload/powerset_merge/powerset_merge.py

```diff
@@ -118,56 +118,67 @@ def merge_powersets(manager: s3_manager.S3Manager) -> None:
+                for match in matches:
+                    match_filename = functions.get_s3_filename(match)
+                    match_timestamped_filename = f".{date_str}.".join(match_filename.split("."))
+                    last_target = (
+                        f"{enums.BucketPath.LAST_VALID.value}/{subbucket_path}/{match_filename}"
+                    )
+                    archive_target = (
+                        f"{enums.BucketPath.ARCHIVE.value}/{subbucket_path}/{match_timestamped_filename}"
+                    )
+                    manager.move_file(
+                        last_target,
+                        archive_target
+                    )
+                    archives.append((archive_target,last_target))
             # otherwise, this is the first instance - after it's in the database,
             # we'll generate a new list of valid tables for the dashboard
             else:
                 is_new_data_package = True
             df = expand_and_concat_powersets(df, latest_path, manager.site)
+            filename=functions.get_s3_filename(latest_path)
             manager.move_file(
-                f"{enums.BucketPath.LATEST.value}/{subbucket_path}",
-                f"{enums.BucketPath.LAST_VALID.value}/{subbucket_path}",
+                f"{enums.BucketPath.LATEST.value}/{subbucket_path}/{filename}",
+                f"{enums.BucketPath.LAST_VALID.value}/{subbucket_path}/{filename}",
             )
 
-            latest_site = site_specific_name.split("/", maxsplit=1)[0]
             manager.update_local_metadata(
-                enums.TransactionKeys.LAST_DATA_UPDATE.value, site=latest_site
+                enums.TransactionKeys.LAST_DATA_UPDATE.value, site=latest_metadata.site
             )
             manager.update_local_metadata(
-                enums.TransactionKeys.LAST_AGGREGATION.value, site=latest_site
+                enums.TransactionKeys.LAST_AGGREGATION.value, site=latest_metadata.site
             )
         except Exception as e:
             manager.error_handler(
                 latest_path,
                 subbucket_path,
                 e,
             )
+            # Undo any archiving we tried to do
+            for archive in archives:
+                manager.move_file(archive[0], archive[1])
             # if a new file fails, we want to replace it with the last valid
             # for purposes of aggregation
-            if any(x.endswith(site_specific_name) for x in last_valid_file_list):
-                df = expand_and_concat_powersets(
-                    df,
-                    f"s3://{manager.s3_bucket_name}/{enums.BucketPath.LAST_VALID.value}"
-                    f"/{subbucket_path}",
-                    manager.site,
-                )
-                manager.update_local_metadata(enums.TransactionKeys.LAST_AGGREGATION.value)
+            if matches := filter(lambda x: subbucket_path in x, last_valid_file_list):
+                for match in matches:
```

Comment:
nit: similar consolidation here to one line: `for match in filter()`. Same "bare `in` string check" comment too.